Code/GraphMol/Kekulize.cpp

```diff
@@ -303,12 +306,26 @@ bool kekulizeWorker(RWMol &mol, const INT_VECT &allAtms,
         // a fairly detailed analysis of all bonds in the molecule to determine
         // which of them is eligible to be converted.
         if (cCand && dBndCands[nbrIdx] &&
-            (mol.getBondBetweenAtoms(curr, nbrIdx)->getIsAromatic() ||
+            (nbrBond->getIsAromatic() ||
              mol.getAtomWithIdx(curr)->getAtomicNum() == 0 ||
              mol.getAtomWithIdx(nbrIdx)->getAtomicNum() == 0)) {
-          opts.push_back(nbrIdx);
+          // in order to try and avoid making wedged bonds double, we will add
+          // this neighbor at the back of the options after this loop if the
+          // bond is wedged. otherwise we append it to the options directly
+          if (nbrBond->getBondDir() == Bond::BondDir::BEGINWEDGE ||
+              nbrBond->getBondDir() == Bond::BondDir::BEGINDASH) {
+            wedgedOpts.push_back(nbrIdx);
+          } else {
+            opts.push_back(nbrIdx);
+          }
         }  // end of curr atoms can have a double bond
       }    // end of looping over neighbors
+      // now append to opts the neighbors connected via wedged bonds,
+      // if any were found
+      for (auto wedgedNbrIdx : wedgedOpts) {
+        opts.push_back(wedgedNbrIdx);
+      }
```

Comment:
I think this can be done without an explicit loop:
```suggestion
      opts.insert(opts.end(), wedgedOpts.begin(), wedgedOpts.end());
```